File: 2019/Python/D1/p2.py
lines = [x[:-1] for x in open("input.txt").readlines()]

def fuel(x):
    f = x//3 - 2
    if f <= 0:
        return 0
    else:
        return f + fuel(f)

def sol():
    ans = 0
    for x in lines:
        ans += fuel(int(x))
    return ans
    
ans = sol()
print(ans)

File: 2019/rust/src/d1/p2.rs
use std::fs;

pub fn main() {
    let data = fs::read_to_string("src/d1/input.txt").expect("read failed");

    let lines: Vec<i64> = data.lines().map(|x| x.parse::<i64>().unwrap()).collect();

    let mut total: i64 = 0;

    for x in lines {
        total += calc(x);
    }

    println!("{}", total);
}

fn calc(n: i64) -> i64 {
    let mut total: i64 = 0;
    let mut x = n;

    loop {
        x = x / 3 - 2;
        if x <= 0 {
            break;
        }
        total += x;
    }

    return total;
}
